holonote/annotate/display.py

```diff
@@ -425,13 +434,24 @@ def get_indices_by_position(self, **inputs) -> list[Any]:
             iter_mask = (
                 (df[f"start[{k}]"] <= v) & (v < df[f"end[{k}]"]) for k, v in inputs.items()
             )
+            subset = reduce(np.logical_and, iter_mask)
+            out = list(df[subset].index)
+        elif self.region_format == "point-point":
+            xk, yk = list(inputs.keys())
+            distance = (df[f"point[{xk}]"] - inputs[xk]) ** 2 + (
+                df[f"point[{yk}]"] - inputs[yk]
+            ) ** 2
+            if (distance > self._nearest_2d_point_threshold**2).all():
+                return []
```

Comment:
This prevents editing and deleting existing annotations, at least in my testing. I'm leaning toward deleting this logic. 

Using your example from: https://github.com/holoviz/holonote/pull/108#issuecomment-2142632247

Reply:
After playing around with it a bit more (and also adding the style options), I can see why this is done. 

I still think it is the correct thing to prioritize the simple example containing "holonote" only, over the more complex one. 

Therefore, I would suggest that `self._nearest_2d_point_threshold` is disabled by default. Also, update the docstring with information that if the value is chosen too small, you can lose the options to edit and delete annotations.  


My style diff:
``` diff
diff --git a/holonote/annotate/display.py b/holonote/annotate/display.py
index 304b42b..27a96c7 100644
--- a/holonote/annotate/display.py
+++ b/holonote/annotate/display.py
@@ -88,12 +88,14 @@ class Style(param.Parameterized):
     line_opts = _StyleOpts(default={})
     span_opts = _StyleOpts(default={})
     rectangle_opts = _StyleOpts(default={})
+    points_opts = _StyleOpts(default={})
 
     # Editor opts
     edit_opts = _StyleOpts(default={"line_color": "black"})
     edit_line_opts = _StyleOpts(default={})
     edit_span_opts = _StyleOpts(default={})
     edit_rectangle_opts = _StyleOpts(default={})
+    edit_points_opts = _StyleOpts(default={})
 
     _groupby = ()
     _colormap = None
@@ -133,6 +135,7 @@ class Style(param.Parameterized):
             hv.opts.HSpans(**opts, **self.span_opts),
             hv.opts.VLines(**opts, **self.line_opts),
             hv.opts.HLines(**opts, **self.line_opts),
+            hv.opts.Points(**opts, **self.points_opts),
         )
 
     def editor(self) -> tuple[hv.Options, ...]:
@@ -148,6 +151,7 @@ class Style(param.Parameterized):
             hv.opts.HSpan(**opts, **self.edit_span_opts),
             hv.opts.VLine(**opts, **self.edit_line_opts),
             hv.opts.HLine(**opts, **self.edit_line_opts),
+            hv.opts.Points(**opts, **self.points_opts),
         )
 
```

Reply:
Thanks for reviewing. I believe the last line of the opts should be `edit_points_opts`? If so, I've implemented that.

> Therefore, I would suggest that self._nearest_2d_point_threshold is disabled by default

I disagree; without it, users click anywhere, and it selects the nearest point, but that can be misleading the users.

https://github.com/holoviz/holonote/assets/15331990/995cb1e3-cd4e-41b8-b5ea-f437471f1414

Based on these comments:

> This prevents editing and deleting existing annotations, at least in my testing.

> I still think it is the correct thing to prioritize the simple example containing "holonote" only, over the more complex one.

We can just up the threshold to 1, and it works fine.

https://github.com/holoviz/holonote/assets/15331990/2e8d2309-aeec-40be-8b5a-413e02a9bbaf

Reply:
> I believe the last line of the opts should be edit_points_opts?

Correct. Copy / paste error.

> We can just up the threshold to 1, and it works fine.

It works "fine" with the current magnitude of `lon`/`lat`. If we edit this, we will see the same problems. 

``` python
factor = 1e3  # or 1e-3
ds["lon"] = ds["lon"] * factor
ds["lat"] = ds["lat"] * factor
``` 

It can be hard to hit on a larger scale. On a smaller scale, we have the same problem as without the limit. So, having it enabled by default, no matter the value, we end up in a Goldilocks situation where it only works if it is just right and creates confusing behavior if it is off.  


https://github.com/holoviz/holonote/assets/19758978/6efcaa44-66b1-499f-a7e2-978cf72c449e



https://github.com/holoviz/holonote/assets/19758978/ec8f8d61-a4d6-4b0d-a8b6-37fcc783e62d



> previous point highlighted instead of where they clicked

This is somewhat improved with the updated style, which highlights the selected annotation.

Reply:
> On a smaller scale, we have the same problem as without the limit. So, having it enabled by default, no matter the value, we end up in a Goldilocks situation where it only works if it is just right and creates confusing behavior if it is off.

Which is why I made it a public param, similar to np.testing.assert_almost_equal

> This is somewhat improved with the updated style, which highlights the selected annotation.

To clarify, I meant while creating a new point, not editing an existing.